File: Easy/0013-Roman-to-Integer/Solution.rb
# @param {String} s
# @return {Integer}
def roman_to_int(s)
    symbol_value = {
        'I' => 1,
        'V' => 5,
        'X' => 10,
        'L' => 50,
        'C' => 100,
        'D' => 500,
        'M' => 1000,
    }
    l = s.length - 1
    result = 0
    i = 0
    while i <= l
        if i == l
            result += symbol_value[s[i]]
            i += 1
        else
            if symbol_value[s[i]] < symbol_value[s[i + 1]]
                result += symbol_value[s[i+1]] - symbol_value[s[i]]
                i += 2
            else
                result += symbol_value[s[i]]
                i += 1
            end
        end
    end
    result
end

File: Easy/0013-Roman-to-Integer/Solution.js
// Time: 8 ms (71.14%), Space: 211.72 MB (10.98%)

/**
 * @param {string} s
 * @return {number}
 */
var romanToInt = function(s) {
    let k = 0;
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] !== val) {
            nums[k] = nums[i];
            k++;
        }
    }
    return k;
};